src/test/java/ir/sahab/kafkaconsumer/EmbeddedKafkaServer.java

```diff
@@ -49,32 +50,44 @@
+        logDir = Files.createTempDirectory("kafka").toFile();
         brokerPort = anOpenPort();
 
-        this.kafkaBrokerConfig.setProperty("zookeeper.connect", zkServer.getAddress());
-        this.kafkaBrokerConfig.setProperty("broker.id", "1");
-        this.kafkaBrokerConfig.setProperty("host.name", localIp);
-        this.kafkaBrokerConfig.setProperty("advertised.host.name", localIp);
-        this.kafkaBrokerConfig.setProperty("port", Integer.toString(brokerPort));
-        this.kafkaBrokerConfig.setProperty("log.dir", logDir.getAbsolutePath());
-        this.kafkaBrokerConfig.setProperty("log.flush.interval.messages", String.valueOf(1));
-        this.kafkaBrokerConfig.setProperty("auto.create.topics.enable", "true");
-        this.kafkaBrokerConfig.setProperty("offsets.topic.replication.factor", "1");
+        // Configs 'host.name' and 'advertised.host.name' are deprecated since kafka version 1.1.
+        // Use 'listeners' and 'advertised.listeners' instead of them. See this:
+        // https://kafka.apache.org/11/documentation.html#configuration
+        kafkaBrokerConfig.setProperty(KafkaConfig.ZkConnectProp(), zkServer.getAddress());
+        kafkaBrokerConfig.setProperty(KafkaConfig.BrokerIdProp(), "1");
+        kafkaBrokerConfig.setProperty(KafkaConfig.ListenersProp(),
+                String.format("PLAINTEXT://%s:%s", localIp, brokerPort));
+        kafkaBrokerConfig.setProperty(KafkaConfig.AdvertisedListenersProp(),
+                String.format("PLAINTEXT://%s:%s", localIp, brokerPort));
+        kafkaBrokerConfig.setProperty(KafkaConfig.PortProp(), Integer.toString(brokerPort));
+        kafkaBrokerConfig.setProperty(KafkaConfig.LogDirProp(), logDir.getAbsolutePath());
+        kafkaBrokerConfig.setProperty(KafkaConfig.LogFlushIntervalMessagesProp(), "1");
+        kafkaBrokerConfig.setProperty(KafkaConfig.AutoCreateTopicsEnableProp(), "true");
+        kafkaBrokerConfig.setProperty(KafkaConfig.OffsetsTopicReplicationFactorProp(), "1");
         this.broker = new KafkaServerStartable(new KafkaConfig(this.kafkaBrokerConfig));
         this.broker.startup();
     }
 
     public void createTopic(String topicName, Integer numPartitions) {
-        String[] arguments = new String[]{"--create", "--zookeeper", zkServer.getAddress(),
-                "--replication-factor", "1",
-                "--partitions", "" + numPartitions,
-                "--topic", topicName};
-        TopicCommandOptions opts = new TopicCommandOptions(arguments);
-        ZkUtils zkUtils = ZkUtils.apply(opts.options().valueOf(opts.zkConnectOpt()),
-                30000, 30000, JaasUtils.isZkSecurityEnabled());
-        logger.info("Executing: CreateTopic " + Arrays.toString(arguments));
-        TopicCommand.createTopic(zkUtils, opts);
+        ZkClient zkClient = null;
+        ZkUtils zkUtils;
+        try {
+            // When not initializing the ZkClient with ZKStringSerializer, createTopic will return without error.
```

Comment:
We have to pass the serializer class to ZkClient constructor. Otherwise createTopic ...